cliqz-logger.js

```diff
@@ -0,0 +1,419 @@
+const LOGGER_VAR_NAME_PATTERN = '__logger_var__';
+// EOL - optional;
+// /*LS-<random number A>*/<greedy matched sources>/*LE-<random number A>*/
+// EOL - optional;
+const IMPORT_PATTERN = new RegExp(
+  `(?:${EOL})?\\/\\*LS-\\d+\\*\\/(?:.|\\s)*?\\/\\*LE-\\d+\\*\\/(?:${EOL})?`);
+// EOL
+// __logger_var__12("3466", "win != maybeAnotherWin");"__logger_var__12";
+// EOL
+const LOG_FUNCTION_CALL_PATTERN = new RegExp(
+  `${EOL}${LOGGER_VAR_NAME_PATTERN}\\d+\\((?:.|\\s)*?\\);"${LOGGER_VAR_NAME_PATTERN}\\d+";`);
+const fs = require('fs');
+const systemPath = require('path');
+
+////////////////////////////////////////////////////////////////////
+// Main method is the starting point.
+// It recieves inputArgs as the only argument.
+const main = function(inputArgs) {
+  let i = 0;
+  // Initial state is 'INJECT' meaning we want to log by default;
+  let state = 'INJECT';
+  // Since we have not gotten any config file path yet
+  // let it be empty string by default;
+  let configPath = '';
+
+  // Default name for logging provider module.
+  // The module should be responsible for what to do with actual logs.
+  // It also has to implement 'init' method which recieves the only
+  // argument as a path to the whole module being logged.
+  // The name will be injected into sources on the first line and then
+  // the next line will call its' 'init' method.
+  let loggerName = 'CliqzLogger';
+  // Default path for logging provider.
+  let loggerPath = 'resource://gre/modules/CliqzLogger.jsm';
+  // Places is an array of file paths which we want to log.
+  // Each path is appended to PWD directory path and read afterwards.
+  // So please fill in places with paths relative to PWD in your system;
+  let places = [];
+  // importAs tells how to import a logging provider.
+  // There might be 2 values: 'browser' and 'es6';
+  // The former is default one and says to import using ChromeUtils;
+  // When the second one is used then a logging provider is imported as an
+  // EcmaScript6 module using 'import' directive;
+  let importAs = 'browser'; // possible values: browser, es6;
+
+  while(inputArgs[i] != null) {
+    if (inputArgs[i] == '--clear') {
+      state = 'CLEAR';
+    } else if (inputArgs[i] == '--config') {
+      // --config option requires a file path afterwards.
+      // If there is not any then we show the error and terminate;
+      if (inputArgs[i + 1] == null) {
+        writeToConsole(
+          `${EOL}Use: node ${process.argv[1]} [--clear] --config <json-config-path>`,
+          `${EOL}${inputArgs[i + 1]} is not defined.`,
+          `${EOL}Terminated.${EOL}`
+        );
+        process.exit();
+        return;
+      }
+
+      // Otherwise we can replace our default config path value with a custom one.
+      configPath = inputArgs[i + 1];
+    } else {
+      // Places also can be augmented through the command line terminal;
+      places.push(inputArgs[i]);
+    }
+
+    i += 1;
+  }
+
+  new Promise(function(resolve, reject) {
```

Comment:
Why create a `Promise` if nothing waits for it?

Reply:
Because it is easier to organize the code inside in that case since reading a file is asynchronous.